service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -112,13 +113,15 @@ public ResponseEntity<RunSetListResponse> getRunSets(UUID methodId, Integer page
     RunSetListResponse response;
 
     if (methodId != null) {
-      List<RunSet> filteredRunSet = Collections.singletonList(runSetDao.getRunSet(methodId));
+      List<RunSet> filteredRunSet =
+          Collections.singletonList(
+              runSetDao.getRunSetWithMethodId(methodId, Optional.ofNullable(pageSize).orElse(10)));
```

Comment:
I'm not 100% sure how singletonList works, but if we're only expecting a single result back, we should probably force the `Optional.ofNullable(pageSize).orElse(10)` to be exactly 1.

Or if we want to allow more, we might want to remove that `singletonList` function.